core/blockchain_reader.go

```diff
@@ -76,6 +76,8 @@ func (bc *BlockChain) GetHeaderByHash(hash common.Hash) *types.Header {
 
 // GetHeaderByNumber retrieves a block header from the database by number,
 // caching it (associated with its hash) if found.
+//
+// This path is commonly accessed when processing eth_getLogs requests.
```

Comment:
Can we remove this comment? I don't think this needs to be here and many API calls hit `GetHeaderByNumber`

Reply:
Fair